#VRML_SIM R2023a utf8
PROTO SimpleRobot [
  field SFVec3f    translation  0 0 0.06
  field SFRotation rotation     0 0 1 0
  field SFString   name         "MyRobot"
  field SFString   controller   "RobotController"
  field MFFloat    battery      [100, 100, 100]
  field SFString   customData   "10,10"
]
{
  Robot {
  translation IS translation
  rotation IS rotation
  name IS name
  controller IS controller
  battery IS battery
  customData IS customData
  scale 3 3 3
children [
    DEF LeftSensorDistance DistanceSensor {
      translation 0.05 0.01 0
      rotation 0 0 1 0.5
      children [
        DEF LeftSensor Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "LeftSensorDistance"
    }
    DEF RightSensorDistance DistanceSensor {
      translation 0.05 -0.01 0
      rotation 0 0 1 -0.5
      children [
        DEF RightSensor Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "RightSensorDistance"
    }
    DEF Compass Compass {
      physics Physics {
      }
    }
    DEF GPS GPS {
    }
    DEF LeftSensor LightSensor {
      translation 0.05 0.01 0
      rotation 0 0 1 0.5
      children [
        DEF LeftSensor Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "LeftSensor"
      colorFilter 0.1 0.1 0.1
    }
    DEF RightSensor LightSensor {
      translation 0.05 -0.01 0
      rotation 0 0 1 -0.5
      children [
        DEF RightSensor Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "RightSensor"
      boundingObject USE RightSensor
      physics Physics {
      }
      colorFilter 0.1 0.1 0.1
    }
    DEF BackLeftWheel HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.03 -0.03 0
      }
      device [
        RotationalMotor {
          name "BackLeftWheel"
          consumptionFactor 0
        }
      ]
      endPoint Solid {
        translation -0.030033560280683096 -0.03012142863822498 4.703586901272662e-05
        rotation 0.9999978879587038 0.001375672738602805 0.0015269586267630066 1.558689156524543
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE Wheel
        physics Physics {
        }
      }
    }
    DEF BackRightWheel HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.03 0.03 0
      }
      device [
        RotationalMotor {
          name "BackRightWheel"
          consumptionFactor 0
        }
      ]
      endPoint Solid {
        translation -0.030040349070045082 0.030100960929338783 -0.00010819864305708715
        rotation 0.9999865094956314 -0.0036846703942624913 -0.00366115157144958 1.5573086451742473
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics Physics {
        }
      }
    }
    DEF FrontRightWheel HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.03 0.03 0
      }
      device [
        RotationalMotor {
          name "FrontRightWheel"
          consumptionFactor 0
        }
      ]
      endPoint Solid {
        translation 0.030028905409587338 0.030123570992669046 9.761269208599574e-05
        rotation 0.9999865096945971 -0.003684643193768846 -0.0036611246021045147 1.5573086921322368
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics Physics {
        }
      }
    }
    DEF FrontLeftWheel HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.03 -0.03 0
      }
      device [
        RotationalMotor {
          name "FrontLeftWheel"
          consumptionFactor 0
        }
      ]
      endPoint Solid {
        translation 0.030040014680815852 -0.030102525474836828 -3.5678186693654256e-05
        rotation 0.9999978880259554 0.0013756515839437353 0.0015269336423774976 1.5586892447686007
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics Physics {
        }
      }
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.02
      }
    }
  ]
  name "MyRobot"
  boundingObject USE Body
  physics Physics {
  }
  controller "RobotController"
  cpuConsumption 0
}
}